test/e2e/scheduling/preemption.go

```diff
@@ -669,6 +669,84 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 			}
 		})
 	})
+
+	ginkgo.Context("PriorityClass endpoints", func() {
+		var cs clientset.Interface
+		f := framework.NewDefaultFramework("sched-preemption-path")
+		var pcs []*schedulingv1.PriorityClass
+
+		ginkgo.BeforeEach(func() {
+			cs = f.ClientSet
+			// Create 2 PriorityClass: p1, p2.
+			for i := 1; i <= 2; i++ {
+				name, val := fmt.Sprintf("p%d", i), int32(i)
+				pc, err := cs.SchedulingV1().PriorityClasses().Create(context.TODO(), &schedulingv1.PriorityClass{ObjectMeta: metav1.ObjectMeta{Name: name}, Value: val}, metav1.CreateOptions{})
+				if err != nil {
+					framework.Logf("Failed to create priority '%v/%v': %v", name, val, err)
+					framework.Logf("Reason: %v. Msg: %v", apierrors.ReasonForError(err), err)
+				}
+				framework.ExpectEqual(err == nil || apierrors.IsAlreadyExists(err), true)
+				pcs = append(pcs, pc)
+			}
+		})
+
+		ginkgo.AfterEach(func() {
+			// Print out additional info if tests failed.
+			if ginkgo.CurrentGinkgoTestDescription().Failed {
+				// list existing priorities
+				priorityList, err := cs.SchedulingV1().PriorityClasses().List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					framework.Logf("Unable to list priorities: %v", err)
+				} else {
+					framework.Logf("List existing priorities:")
+					for _, p := range priorityList.Items {
+						framework.Logf("%v/%v created at %v", p.Name, p.Value, p.CreationTimestamp)
+					}
+				}
+			}
+
+			// Cannot run collection deletion which would delete all system level priority classes.
+			for _, pc := range pcs {
+				err := cs.SchedulingV1().PriorityClasses().Delete(context.TODO(), pc.Name, *metav1.NewDeleteOptions(0))
+				framework.ExpectNoError(err)
+			}
+		})
+
+		ginkgo.It("verify PriorityClass endpoints can be operated with different HTTP methods", func() {
+			// 1. Patch/Update on immutable fields will fail.
```

Comment:
even "get" and "list" have been implicitly included in this testcase, I am wondering whether we need to explicitly test it here?

Reply:
I think "get" is explicitly covered? (in the for loop in step 3).

"list" does need to be explicitly tested, I will get it covered.

Reply:
That's as expected. As PriorityClass is a cluster-level resource, if we go delete the collection, it would delete all PriorityClass - which includes the system-critical ones, which is not accepted.

I also mentioned it in the PR description (check out the "Special notes for your reviewer:" section).

Reply:
I would recommend using labels selectors to delete collections, make sure that the resources managed in this test are statically labeled, so at the end, they can be deleted using a given label. e.g:

```go
_ = f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "test-rc-static=true"})
```

https://github.com/kubernetes/kubernetes/pull/90880/files#diff-2a2edebd256ce20f5b0964acb9360c53d8922c6ed84ab85d60f11649503f5b70R381

Reply:
> I would recommend using labels selectors to delete collections, make sure that the resources managed in this test are statically labeled, so at the end, they can be deleted using a given label. e.g:
> 
> ```go
> _ = f.ClientSet.CoreV1().ReplicationControllers(testRcNamespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "test-rc-static=true"})
> ```
> 
> https://github.com/kubernetes/kubernetes/pull/90880/files#diff-2a2edebd256ce20f5b0964acb9360c53d8922c6ed84ab85d60f11649503f5b70R381

@Huang-Wei please reach out if you need more support to get the `deleteSchedulingV1CollectionPriorityClass` endpoint tested. 
Thank you for the great work.

Reply:
Thanks @BobyMCbobs @Riaankl I will submit another PR exercising `DeleteCollection` with ListOptions.